### DISCUSSION TITLE
hires upscaler - automatic?

### DISCUSSION THREAD

**Discussion by rbfussell** [AIrjen/OneButtonPrompt#86] [Q&A] [answered]
What exactly does this use to "automatically" determine the upscaler?  Does this just mean, just choose one randomly,  or does it use prompt context to choose based on assumed image type?

**Comment by AIrjen** [comment, marked as answer]
Hi! Yes, it tries but it is very basic in the current implementation. It defaults to ultrasharp (or R-ERSGRAN if you dont have it). It switches to R-ESGRAN ANIME, based on certain words in the prompt. It also sets the denoising strength.

this was something I made when it was still a personal project.
